### PR TITLE
DM-34589: Add a way to mark which dimensions populate others.

### PR DESCRIPTION
**Pull request by TallJimbo**
In the future, this will be used to make it easier to export dimension data without knowing the details of the dimension combinations that represent many-to-main joins (DM-34838).

There's no real gain to updating the dimensions configuration before that happens, but merging these changes early is useful because it will make software versions with just this change much better able to handle future data repositories that use them in their dimensions configuration, even if the associated functionality isn't available as a result.

## Checklist

- [x] ran ~Jenkins~ local lsstsw
- [x] added a release note for user-visible changes to `doc/changes`

### CODE REVIEW

**Review by timj — COMMENTED** [review]
I don't understand why this can't have any tests? Why can't we have a small test dimensions universe that has a couple of elements in where one populates the other. We have special test dimensions.yaml content in pipe_base and ctrl_mpexec.  Why can't there be one call to get_elements_populated_by in one of the dimensions tests?

**Comment by TallJimbo** [comment]
I can add some tests with a custom dimensions universe later in the week.

**Comment by TallJimbo** [comment]
I think I'm changing my mind (a bit) again on how far to go on this ticket - I'm thinking now that we _should_ update the default dimensions config to use `populated_by`, even if it doesn't do anything yet, because that will hopefully save repos created between now and some date in the future from needing a migration.  But I still don't think we want to migrate any existing repos to use it.

So, since I think the rest of the migration changes are pretty close to merging over on DM-33942, let's hold this ticket up until that merges, then update the default dimensions config on this ticket after rebasing (and add a test based on that default config then).

**Comment by timj** [comment]
#674 (DM-33942) was merged.

File: doc/changes/DM-34589.misc.md
Comment (timj) [inline doc/changes/DM-34589.misc.md]:
```suggestion
Add dimensions config entries that declare that the `visit` dimension "populates" various dimension elements that define many-to-many relationships.
```
?

**Review by timj — APPROVED** [review]

**Pull request merged**
